src/StoragelessSession/Http/SessionMiddleware.php

```diff
@@ -49,96 +66,191 @@
     private $tokenParser;
 
     /**
-     * @param Signer $signer
-     * @param Key $signatureKey
-     * @param Key $verificationKey
-     * @param Parser $tokenParser
+     * @var SetCookie
+     */
+    private $defaultCookie;
+
+    /**
+     * @param Signer    $signer
+     * @param string    $signatureKey
+     * @param string    $verificationKey
+     * @param SetCookie $defaultCookie
+     * @param Parser    $tokenParser
+     * @param int       $expirationTime
      */
     public function __construct(
         Signer $signer,
-        Key $signatureKey,
-        Key $verificationKey,
-        Parser $tokenParser
+        string $signatureKey,
+        string $verificationKey,
+        SetCookie $defaultCookie,
+        Parser $tokenParser,
+        int $expirationTime
     ) {
-        $this->signer = $signer;
-        $this->signatureKey = $signatureKey;
+        $this->signer          = $signer;
+        $this->signatureKey    = $signatureKey;
         $this->verificationKey = $verificationKey;
-        $this->tokenParser = $tokenParser;
+        $this->tokenParser     = $tokenParser;
+        $this->defaultCookie   = clone $defaultCookie;
+        $this->expirationTime  = $expirationTime;
+    }
+
+    /**
+     * This constructor simplifies instantiation when using HTTPS (REQUIRED!) and symmetric key encription
+     *
+     * @param string $symmetricKey
+     * @param int    $expirationTime
+     *
+     * @return self
+     */
+    public static function fromSymmetricKeyDefaults(string $symmetricKey, int $expirationTime)
+    {
+        return new self(
+            new Signer\Hmac\Sha256(),
+            $symmetricKey,
+            $symmetricKey,
+            SetCookie::create(self::DEFAULT_COOKIE)
+                ->withSecure(true)
+                ->withHttpOnly(true),
+            new Parser(),
+            $expirationTime
+        );
+    }
+
+    /**
+     * This constructor simplifies instantiation when using HTTPS (REQUIRED!) and asymmetric key encription
+     * based on RSA keys
+     *
+     * @param string $privateRsaKey
+     * @param string $publicRsaKey
+     * @param int    $expirationTime
+     *
+     * @return self
+     */
+    public static function fromAsymmetricKey(string $privateRsaKey, string $publicRsaKey, int $expirationTime)
+    {
+        return new self(
+            new Signer\Rsa\Sha256(),
+            $privateRsaKey,
+            $publicRsaKey,
+            SetCookie::create(self::DEFAULT_COOKIE)
+                ->withSecure(true)
+                ->withHttpOnly(true),
+            new Parser(),
+            $expirationTime
+        );
     }
 
     /**
      * {@inheritdoc}
+     *
+     * @throws \InvalidArgumentException
+     * @throws \OutOfBoundsException
+     * @throws \BadMethodCallException
      */
     public function __invoke(Request $request, Response $response, callable $out = null)
```

Comment:
![yes-we-can](https://cloud.githubusercontent.com/assets/1850941/10708367/4fff6632-79e4-11e5-8154-bdc242acbb26.jpg)